### DISCUSSION TITLE
How can I combine this text without a space inbetween in MessageBuilder?

### DISCUSSION THREAD

**Discussion by karl-police** [Chatterino/chatterino2#3500] [Q&A] [answered]
Working on something and trying to figure out an issue here:

![](https://i.imgur.com/waPPrCd.png)

Messages are being combined with ``builder.emplace<TextElement>``

I probably need to look more in the source code to see what I can do, but for now, I don't know.

Does someone know?

I am trying to combine the bolded message with the other message, just that it shouldn't have a space after "Suspicious User"

**Comment by fourtf** [comment, marked as answer]
Check `hasTrailingSpace` in MessageElement https://github.com/Chatterino/chatterino2/blob/820099821a109ae855b1fd9fe327e919af41bc8c/src/messages/MessageElement.hpp#L162

Edit: and `setTrailingSpace` i suppose

**Reply by karl-police** [reply]
> Check `hasTrailingSpace` in MessageElement
> 
> https://github.com/Chatterino/chatterino2/blob/820099821a109ae855b1fd9fe327e919af41bc8c/src/messages/MessageElement.hpp#L162
> 
> Edit: and `setTrailingSpace` i suppose

This does lead to somewhere, unfortunately not how I expected it would do it. It removes the spaces within the entire message.
There could be a way to use this in a smart way, but.

I would probably have to make it build the text part "Suspicious" and then "User" but have the setTrailingSpace there on false, and then combine it with the rest

does that sound good?

**Reply by karl-police** [reply]
> > Check `hasTrailingSpace` in MessageElement
> > https://github.com/Chatterino/chatterino2/blob/820099821a109ae855b1fd9fe327e919af41bc8c/src/messages/MessageElement.hpp#L162
> > 
> > Edit: and `setTrailingSpace` i suppose
> 
> This does lead to somewhere, unfortunately not how I expected it would do it. It removes the spaces within the entire message. There could be a way to use this in a smart way, but.
> 
> I would probably have to make it build the text part "Suspicious" and then "User" but have the setTrailingSpace there on false, and then combine it with the rest
> 
> does that sound good?

see 
![image](https://user-images.githubusercontent.com/12023782/149334305-a20dd682-9bb5-4757-b89d-281c362b2a0d.png)

**Reply by fourtf** [reply]
Looks good to me 👍

**Reply by karl-police** [reply]
> Looks good to me 👍

Thanks, for your help.

Do you want to help me with this too? https://github.com/Chatterino/chatterino2/discussions/3499

**Reply by fourtf** [reply]
Not rn
